### PR TITLE
Add acceptance tests

### PR DESCRIPTION
**Pull request by taras**
This branch is rebased on EmberCLI 0.2.7 branch.

## Includes
- Acceptance Tests
   - Simple Test
   - Configurable Columns Test
   - Removable Columns Test
- Refactored template substitution with `ember-code-snippet`
- Refactored all examples to use new `{{code-snippet}}`
- Added random module with `randomNumber` and `randomDate` that produce predicable values based on seed value
- Added `rowText` test helper for getting cell text for a row

### CODE REVIEW

**Comment by taras** [comment]
@azirbel I rebased this branch against #354

File: tests/dummy/app/components/configurable-table-source.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
```
Comment (azirbel) [inline tests/dummy/app/components/configurable-table-source.js]: do you really want to make this a component? it's not needed after the changes on master

Comment (taras) [inline tests/dummy/app/components/configurable-table-source.js]: No, I don't like it. let me see if I can revert it.

**Comment by taras** [comment]
I reverted that change

File: tests/dummy/app/templates/ajax.hbs [hunk 2]
```diff
@@ -5,7 +5,17 @@
     <div class="col-md-12">
       <div class="example-container">
         <div class="ember-table-example-container">
-          {{partial "ajax-table/ajax-table"}}
+          {{!// BEGIN-SNIPPET ajax-table}}
```
Comment (azirbel) [inline tests/dummy/app/templates/ajax.hbs]: YESSS

File: tests/helpers/ember-table.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import Ember from 'ember';
+
+Ember.Test.registerHelper('rowText', function rowText(app, rowIndex=0){
+  let cells = app.$(`.ember-table-table-row:eq(${rowIndex}) .ember-table-cell`);
```
Comment (azirbel) [inline tests/helpers/ember-table.js]: why backticks?

File: tests/dummy/app/utils/random.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+var seed = 6;
+// source: http://indiegamr.com/generate-repeatable-random-numbers-in-js/
+// in order to work 'Math.seed' must NOT be undefined,
+// so in any case, you HAVE to provide a Math.seed
+export function randomNumber(max = 1, min = 0, randomSeed = undefined) {
```
Comment (azirbel) [inline tests/dummy/app/utils/random.js]: why add `randomSeed` as the parameter here? I think it should be done separately, as it is currently done, via `setRandomSeed`.

Comment (azirbel) [inline tests/dummy/app/utils/random.js]: never mind - now I see the usage a few lines down.

File: tests/acceptance/configurable-columns-test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    Ember.run(application, 'destroy');
+  }
+});
+
+test('table renders', function(assert) {
+  visit('/configurable-columns');
+  andThen(function() {
+    assert.equal(currentPath(), 'configurable-columns');
+    assert.deepEqual(rowText(0), ['Date', 'Open', 'High', 'Low', 'Close'], "headers");
+    assert.deepEqual(rowText(1), ['Thu Jul 14 2005', '22.49', '-18.01', '-47.10', '-0.30'], "values in first row");
```
Comment (azirbel) [inline tests/acceptance/configurable-columns-test.js]: break into two lines

File: tests/acceptance/removable-columns-test.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    Ember.run(application, 'destroy');
+  }
+});
+
+test('table renders', function(assert) {
+  visit('/removable-columns');
+  andThen(function() {
+    assert.equal(currentPath(), 'removable-columns');
+    assert.deepEqual(rowText(0), ['Date', 'Open', 'High', 'Low', 'Close'], "headers");
+    assert.deepEqual(rowText(1), ['Thu Jul 14 2005', '22.49', '-18.01', '-47.10', '-0.30'], "values in first row");
```
Comment (azirbel) [inline tests/acceptance/removable-columns-test.js]: break some of these into 2 lines

**Comment by azirbel** [comment]
This is so awesome, @taras. LGTM after a couple recent comments about line length.

It's too bad that we have to use `randomNumber`, because it makes the examples not work when copy+pasted. But I think it's worth it to make things testable. We can change that later if we change our minds. Another alternative would be to use more predictable content (ascending numbers?).

Could you squash into a single commit and make sure `dist` files are updated? Then I'll merge.

File: tests/helpers/ember-table.js (hunk 3 above)
Comment (taras) [inline tests/helpers/ember-table.js]: Backticks are ES6 template strings that are transpiled by babel.js to ```'.ember-table-table-row:eq(' + rowIndex + ') .ember-table-cell'

**Comment by taras** [comment]
@azirbel I think we can stub Math.random in tests and go back to use it. I'm not sure about Date because there is no Date.random. With that said, I'm not sure how much we should focus on making the examples copypastable especially the `tableContent`. 

I think for the benefit of people who're going to start using Ember Table, we should try to make the examples look as close to real life as possible, include refactoring the examples to use routes to provide data rather than controllers.

I applied the line changes that you suggested, regenerated the dist files(which didn't change cause I regenerated them previously) and squashed into 1 commit.

Should be good to go.
